sgkit/stats/aggregation.py

```diff
@@ -285,6 +285,143 @@ def call_rate(ds: Dataset, dim: Dimension, call_genotype_mask: Hashable) -> Data
     )
 
 
+@numba_jit
+def _greatest_common_denominator(x: int, y: int) -> int:
+    while y != 0:
+        t = x % y
+        x = y
+        y = t
+    return x
+
+
+# TODO: replace with math.comb when supported by numba
+@numba_jit
+def _comb(n: int, k: int) -> int:
+    if k > n:
+        return 0  # pragma: no cover
+    r = 1
+    for d in range(1, k + 1):
+        gcd = _greatest_common_denominator(r, d)
+        r //= gcd
+        r *= n
+        r //= d // gcd
+        n -= 1
+    return r
+
+
+@numba_jit
+def _comb_with_replacement(n: int, k: int) -> int:
```

Comment:
If you want to get into optimisation here, I suspect a small lookup table covering the most common values of n and k here, with a fallback to the general implementation would be very fast.

Reply:
I couldn't quite get it working efficiently, maybe how I'm storing the table? Have specialized for the biallelic case instead which is a massive improvement (see below).

Reply:
Couldn't the lookup just be a global array with say values of `n` in the rows and values of `k` in the columns with values equal to whatever `_comb` outputs there?  I'm curious what happened when you tried.

Given that `_comb` is O(k log(k)) it's probably not a big deal but I could see that possibly making a meaningful difference with high ploidy data.

Reply:
Thanks, now seeing 3-5x speedup for multi-allelic cases tested above. Don't know what I was doing last time 🤷.

Reply:
Great!